### ISSUE TITLE
Unexpected trailing comma after rest operator

### ISSUE THREAD

**Issue by FezVrasta** [eslint/eslint#8273]
**Tell us about your environment**

* **ESLint Version:** v3.8.1
* **Node Version:** v7.5.0
* **npm Version:** 4.1.2

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-ESLint

**Please show your full configuration:**

```
{
  root: true,
  extends: 'eslint:recommended',
  env: {
      browser: true,
      node: true,
      es6: true,
      jest: true,
  },
  parser: 'babel-eslint',
  parserOptions: {
    ecmaVersion: 6,
    sourceType: 'module',
    ecmaFeatures: {
      experimentalObjectRestSpread: true,
      jsx: true,
    },
  },
  rules: {
    'comma-dangle': ['error', 'always-multiline'],
  },
  plugins: [
    'react',
  ],
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
function Component({
  foo,
  bar,
  ...props,
}) {
  return foo + bar;
}
```

**What did you expect to happen?**

The linter shouldn't complain about the trailing comma after `...props`

**What actually happened? Please include the actual, raw output from ESLint.**
```
error  Unexpected trailing comma  comma-dangle
```
Is thrown on the `...props` line.

If I don't use a rest operator, the linting works just fine.

**Comment by FezVrasta** [comment]
So, actually I was able to reproduce the bug on eslint.org/demo

1. Source Type: Module
2. Paste this code inside:

```
/*eslint comma-dangle: ["error", "always-multiline"]*/
export default function A({
  a,
  b,
  ...c,
}) {
  return { a, b, ...c};
}
```

Note that the `...c` will throw an error.
If you remove the 3 dots from `c`, it will work as expected.

**Comment by mysticatea** [comment]
Thank you for this issue.

However, this is the correct behavior.
[The language spec](https://github.com/sebmarkbage/ecmascript-rest-spread) disallows the trailing comma preceded by rest properties.

**Comment by FezVrasta** [comment]
Oh, I see. Thank you for the info, I guess it makes sense actually.
